zerver/lib/actions.py

```diff
@@ -1337,9 +1337,7 @@ def compute_jabber_user_fullname(email: str) -> str:
     return email.split("@")[0] + " (XMPP)"
 
 
-@cache_with_key(
-    lambda realm, email, f: user_profile_by_email_cache_key(email), timeout=3600 * 24 * 7
-)
+@cache_with_key(lambda realm, email, f: user_profile_cache_key(email, realm), timeout=3600 * 24 * 7)
```

Comment:
I'm not sure if this was even correct before? Why was this caching only based on `email`, and disregarding the realm?

Reply:
Yeah, this is likely improperly converted code from the migration to add subdomains.  

I suspect it has limited impact because the Zephyr realm is the only one on zulip.com where it's possible to do mirroring.

Let's extract this fix as a separate prep commit.